Tests/Shared.Specs/AssertionExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Threading.Tasks;
+using FluentAssertions.Common;
+using FluentAssertions.Specialized;
+
+namespace FluentAssertions.Specs
+{
+    internal static class AssertionExtensions
+    {
+        private static readonly AggregateExceptionExtractor extractor = new AggregateExceptionExtractor();
+
+        public static NonGenericAsyncFunctionAssertions Should(this Func<Task> action, ITimer timer)
+        {
+            return new NonGenericAsyncFunctionAssertions(action, extractor, timer);
```

Comment:
We're not wrapping `action` in `ExecuteInDefaultSynchronizationContext`.

Reply:
You mean here, in the constructor?

Reply:
Yes, we currently use `ExecuteInDefaultSynchronizationContext` in three extensions methods in `AssertionExtensions`.
But I have been thinking about moving the wrapping closer to calls to `Wait()`.
E.g. to null check the action before invoking it.

Reply:
This is the unit test project. That `ExecuteInDefaultSynchronizationContext` is not available there.

Reply:
okay.